mne/utils/check.py

```diff
@@ -562,6 +562,43 @@ def _check_depth(depth, kind='depth_mne'):
     return _handle_default(kind, depth)
 
 
+def _check_dict_keys(user_options, allowed_options,
+                     user_opt_name="Channel name(s)", validator_name="info"):
+    """Check that the keys in dictionary are valid against a set list.
+
+    Return the input dictionary if it is valid,
+    otherwise raise a ValueError with a readable error message.
+
+    Parameters
+    ----------
+    user_options : dict
+        The name of the parameter to check. This is used in the error message.
+    allowed_options : list
+        All possible valid key names.
+    user_opt_name : str
+        Meaningful name of the items stored in the input dictionary.
+    validator_name : str
+        Meaningful name of the values against which the keys are
+        validated against.
```

Comment:
this doesn't look right / I find this hard to understand.  Suggestion:
```suggestion
    mapping : dict
        The user-provided dict whose keys we want to check.
    valid_keys : iterable
        The valid keys.
    key_description : str
        Description of the keys in ``mapping``, e.g., "channel name(s)" or
        "annotation(s)".
    valid_key_source : str
        Description of the ``valid_keys`` source, e.g., "info dict" or
        "annotations in the data".
```

Note: `valid_keys` could even be a dict, and it would still work. No need to restrict to `list`.